rt/ws/security/src/main/java/org/apache/cxf/ws/security/trust/STSClient.java

```diff
@@ -87,6 +93,76 @@ public SecurityToken requestSecurityToken(
         }
         return token;
     }
+    
+    /**
+     * see WS-Trust 1.4 chapter 8:
+     * http://docs.oasis-open.org/ws-sx/ws-trust/v1.4/errata01/os/ws-trust-1.4-errata01-os-complete.html#_Toc325658962
+     * 
+     * Creating of (manually triggered second) request with a
+     * RequestSecurityTokenResponse Object
+     * <wst:RequestSecurityTokenResponse xmlns:wst=
+     * "http://docs.oasis-open.org/ws-sx/ws-trust/200512" Context=
+     * "d956dafc-1da5-4661-bab7-834640e659ec"> <wst:SignChallengeResponse>
+     * <wst:Challenge>4451658898</wst:Challenge> </wst:SignChallengeResponse>
+     * </wst:RequestSecurityTokenResponse>
+     * 
+     * @param action
+     * @param challengeValue the challenge value that was received by first
+     *                       issue-response
+     * @return SecurityToken
+     * @throws Exception
+     */
+    public SecurityToken requestSecurityTokenResponse(String action, String challengeValue) throws Exception {
```

Comment:
I think I'd prefer to remove this method altogether. Instead, refactor "issue" so that it can work for both cases. The only difference should be "RequestSecurityTokenResponse" instead of "RequestSecurityToken" for the normal issue case. Maybe the writing out of the Challenge itself could be delegated to the Challenge class itself.

Then in SecurityToken.requestSecurityToken, I think we could handle the SignChallenge case automatically. If the SecurityToken that is returned from the STS contains a SignChallenge then invoke again on the STS with the challenge.